test/position-managers/PositionManager.t.sol

```diff
@@ -0,0 +1,263 @@
+        uint256 balance1Before = currency1.balanceOfSelf();
+
+        uint256 tokenId = lpm.nextTokenId();
+        BalanceDelta delta = mint(range, liquidityToAdd, address(this), ZERO_BYTES);
+
+        assertEq(tokenId, 1);
+        assertEq(lpm.ownerOf(tokenId), address(this));
+
+        bytes32 positionId =
+            keccak256(abi.encodePacked(address(lpm), range.tickLower, range.tickUpper, bytes32(tokenId)));
+        (uint256 liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, uint256(params.liquidityDelta));
+        assertEq(balance0Before - currency0.balanceOfSelf(), uint256(int256(-delta.amount0())), "incorrect amount0");
+        assertEq(balance1Before - currency1.balanceOfSelf(), uint256(int256(-delta.amount1())), "incorrect amount1");
+    }
+
+    function test_mint_exactTokenRatios() public {
+        int24 tickLower = -int24(key.tickSpacing);
+        int24 tickUpper = int24(key.tickSpacing);
+        uint256 amount0Desired = 100e18;
+        uint256 amount1Desired = 100e18;
+        uint256 liquidityToAdd = LiquidityAmounts.getLiquidityForAmounts(
+            SQRT_PRICE_1_1,
+            TickMath.getSqrtPriceAtTick(tickLower),
+            TickMath.getSqrtPriceAtTick(tickUpper),
+            amount0Desired,
+            amount1Desired
+        );
+
+        LiquidityRange memory range = LiquidityRange({poolKey: key, tickLower: tickLower, tickUpper: tickUpper});
+
+        uint256 balance0Before = currency0.balanceOfSelf();
+        uint256 balance1Before = currency1.balanceOfSelf();
+
+        uint256 tokenId = lpm.nextTokenId();
+        BalanceDelta delta = mint(range, liquidityToAdd, address(this), ZERO_BYTES);
+
+        uint256 balance0After = currency0.balanceOfSelf();
+        uint256 balance1After = currency1.balanceOfSelf();
+
+        assertEq(tokenId, 1);
+        assertEq(lpm.ownerOf(1), address(this));
+
+        assertEq(uint256(int256(-delta.amount0())), amount0Desired);
+        assertEq(uint256(int256(-delta.amount1())), amount1Desired);
+        assertEq(balance0Before - balance0After, uint256(int256(-delta.amount0())));
+        assertEq(balance1Before - balance1After, uint256(int256(-delta.amount1())));
+    }
+
+    function test_mint_recipient(IPoolManager.ModifyLiquidityParams memory seedParams) public {
+        IPoolManager.ModifyLiquidityParams memory params = createFuzzyLiquidityParams(key, seedParams, SQRT_PRICE_1_1);
+        uint256 liquidityToAdd =
+            params.liquidityDelta < 0 ? uint256(-params.liquidityDelta) : uint256(params.liquidityDelta);
+
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+
+        uint256 tokenId = lpm.nextTokenId();
+        mint(range, liquidityToAdd, address(alice), ZERO_BYTES);
+
+        assertEq(tokenId, 1);
+        assertEq(lpm.ownerOf(tokenId), alice);
```

Comment:
check that the person whose balance changed - i.e. the person who paid for the minting, was not `alice`, but the caller